nautobot/extras/api/views.py

```diff
@@ -143,6 +145,22 @@ class GitRepositoryViewSet(CustomFieldModelViewSet):
     serializer_class = serializers.GitRepositorySerializer
     filterset_class = filters.GitRepositoryFilterSet
 
+    @swagger_auto_schema(method="post", request_body=serializers.GitRepositorySerializer)
+    @action(detail=True, methods=["post"])
+    def sync(self, request, pk):
+        """
+        Enqueue pull git repository and refresh data.
+        """
+        if not request.user.has_perm("extras.change_gitrepository"):
+            raise PermissionDenied("This user does not have permission to make changes to Git repositories.")
+
+        if not Worker.count(get_connection("default")):
```

Comment:
@glennmatthews would it make sense to break out another queue for git?

Reply:
As implemented, a Git sync operation works very much like a Job execution, including the creation of and logging to a JobResult object. If we want to continue down the direction of having a separate queue for every distinct "category" of background workflow, would it make sense to move Jobs into a distinct "jobs" queue as well and leave the "default" queue unused by default?

Reply:
Related to #245 in that in adding `django-redis` we have to introduce the `CACHES` Django setting. We can [configure RQ to read from that](https://github.com/rq/django-rq#support-for-django-redis-and-django-redis-cache), which will simplify configuration and adding new queues less tedious.

I'll get that wrapped here shortly.

Reply:
How do we want to proceed here - can we get this featurette merged with the existing queueing for jobs and git syncs, and open a feature request to split them out into distinct queues?

Reply:
Filed https://github.com/nautobot/nautobot/issues/345 to track this idea.